skbeam/core/accumulators/binned_statistic.py

```diff
@@ -218,6 +219,14 @@ def flatcount(self):
             self._flatcount = np.bincount(self.xy, None)
         return self._flatcount
 
+    @property
+    def argsort_index(self):
+        # Compute argsort the first time it is accessed. Some statistics
+        # never access it.
+        if self._argsort_index is None:
+            self._argsort_index = self.xy.argsort()
```

Comment:
Is there a usage pattern where the user changes ``xy``? If so we'll have a cache invalidation problem here. From my brief read of this class it seems that  ``xy`` is only set in the ``__init__`, so if the user mucks with ``xy`` that could cause problems, but I'm also not sure that we can do anything to prevent that other than document it. :shrug:

Reply:
We'd need cache invalidation on the `flatcount` too. If they are well versed enough to play with `xy` then I'd hope that they understand what that means in terms of the internals. Maybe leave this for another PR?

Reply:
Yeah I'm not proposing any new work here. I was curious about the thinking on cache invalidation. Apologies for not being clearer